addons/website_blog/models/website_blog.py

```diff
@@ -246,6 +247,8 @@ def create(self, cr, uid, vals, context=None):
             vals['content'] = self._postproces_content(cr, uid, None, vals['content'], context=context)
         create_context = dict(context, mail_create_nolog=True)
         post_id = super(BlogPost, self).create(cr, uid, vals, context=create_context)
+        post = self.browse(cr, uid, post_id, context)
+        post.date_published = post.create_date
```

Comment:
@rga-odoo ,
Need clone of create_date for date_publish, Never find create_date before super call.